libs/langchain/langchain/retrievers/document_compressors/chain_filter.py

```diff
@@ -45,14 +46,49 @@ def compress_documents(
     ) -> Sequence[Document]:
         """Filter down documents based on their relevance to the query."""
         filtered_docs = []
-        for doc in documents:
-            _input = self.get_input(query, doc)
-            output_dict = self.llm_chain.invoke(_input, config={"callbacks": callbacks})
+
+        config = RunnableConfig(callbacks=callbacks)
```

Comment:
Might be worthwhile to update the docs to say that this runs without a limit on the max concurrency. This could lead to issues for some users that will be difficult to debug.

We'll need to make a change here to expose `config` in the signature instead of callbacks, but that's not for this PR